gems/smithy/lib/smithy/templates/client/auth_plugin.erb

```diff
@@ -20,19 +20,19 @@ module <%= module_name %>
       option(
         :auth_schemes,
         doc_type: Hash,
-        rbs_type: 'Hash[String, Smithy::Client::AuthScheme]',
+        rbs_type: 'Hash[String, String]',
```

Comment:
I think this should map string to identity provider, and values will be populated by plugins.